### ISSUE TITLE
Freeze after some commands after upgrade to 0.67.0 

### ISSUE THREAD

**Issue by azzamsa** [nushell/nushell#6339]
### Describe the bug

I didn't face this behavior in [0.66.2](https://github.com/nushell/nushell/releases/tag/0.66.2). Now after upgrade to 0.67.0 I got my terminal freeze after some commands, I need to span new terminal because of this. Pressing `Ctrl-c` or `Ctrl-d` didn't work.

### How to reproduce

1. Install 0.67.0 
2. Just do your normal activity 
3. After some times it freeze (no matter what commands you invoke previously)

### Expected behavior

To run the commands and exits succesfully

### Screenshots

![image](https://user-images.githubusercontent.com/17734314/185053923-4ceccefd-54cd-4b28-ad5f-44e8bb10ca8e.png)

### Configuration

| key                | value                               |
| ------------------ | ----------------------------------- |
| version            | 0.67.0                              |
| branch             |                                     |
| commit_hash        |                                     |
| build_os           | linux-x86_64                        |
| build_target       | x86_64-unknown-linux-gnu            |
| rust_version       | rustc 1.63.0 (4b91a6ea7 2022-08-08) |
| rust_channel       | stable-x86_64-unknown-linux-gnu     |
| cargo_version      | cargo 1.63.0 (fd9c4297c 2022-07-01) |
| pkg_version        | 0.67.0                              |
| build_time         | 2022-08-17 12:51:50 +07:00          |
| build_rust_channel | release                             |
| features           | default, trash, which, zip          |
| installed_plugins  |                                     |

### Additional context

❯ macchina

  Kernel   -  5.10.0-16-amd64
  Distro   -  Debian GNU/Linux 11 (bullseye)
  DE       -  GNOME
  WM       -  GNOME Shell (X11)
  CPU      -  Intel® Core™ i5-7200U CPU @ 2.50GHz (4)
  Memory   -  3.7 GB/12.2 GB

**Comment by hustcer** [comment]
Works well for me on mac

**Comment by azzamsa** [comment]
Any suggestions how to debug this issue?
I always got random freeze till now.

**Comment by azzamsa** [comment]
After daily usage, I find the cause. It causes by pnpm/eleventy. The "freeze" only occurred whenever I work with those projects.

**Comment by rgwood** [comment]
I think this is worth keeping open until we solve it. I've had a few similar freezes.

**Comment by azzamsa** [comment]
> I think this is worth keeping open until we solve it. I've had a few similar freezes.

Yes, the strange thing is that I don't have any similar issue working with the same project in Fish shell.

**Comment by fdncred** [comment]
if it's project based, are you using something like gstat in your prompt? I could see how there could be a bug in getting git statuses that could cause problems.

**Comment by azzamsa** [comment]
This is the command  that occasionally freeze my prompt.

```diff
         "dev:styles": "parcel watch 'src/styles/main.scss' --dist-dir 'dist/styles' --public-url './'",
         "dev:scripts": "parcel watch 'src/scripts/main.js' --dist-dir 'dist/scripts'",
-        "dev:app": "pnpm run dev:styles & pnpm run dev:scripts & eleventy --serve",
-        "dev": "pnpm run dev:app",
+        "dev:app": "eleventy --serve",
+        "dev": "run-p dev:styles dev:scripts dev:app",
```

I thought it because the use of `&`, so I migrated to `run-p` (npm-run-all), But no luck.
I use this prompt https://github.com/azzamsa/dotfiles/blob/61a68518f545f92df690aecb8802681ce21fa0e2/starship/.config/starship.toml

With the same exact prompt config and project, Fish shell works fine. (previously). I never switched to Fish again after installing Nu last week.

**Comment by evelant** [comment]
I'm seeing similar things. On macos arm64 0.69.1, when running npm scripts a lot of the time ctrl+c does not fully exit the command and hangs nushell. I am also using pnpm so perhaps pnpm is doing something to trigger this bug?

**Comment by azzamsa** [comment]
When using ZSH. It looks like this.

```shell
[11ty] Copied 165 files / Wrote 93 files in 1.45 seconds (15.6ms each, v1.0.2)
[11ty] Watching…
[Browsersync] Reloading Browsers...                                                                                                               Error [ERR_SERVER_NOT_RUNNING]: Server is not running.
    at new NodeError (node:internal/errors:393:5)
    at Server.close (node:net:1827:12)
    at Object.onceWrapper (node:events:627:28)
    at Server.emit (node:events:513:28)
    at emitCloseNT (node:net:1880:8)
    at process.processTicksAndRejections (node:internal/process/task_queues:81:21) {
  code: 'ERR_SERVER_NOT_RUNNING'
}
Parcel is shutting down...
^Cmake: *** [Makefile:21: dev] Interrupt


user.com on  master is 📦 v5.11.0 via  v18.8.0 took 8h18m16s
❯ Error: setRawMode EIO
    at ReadStream.setRawMode (node:tty:77:31)
    at process.exit (/home/user/projects/user.com/node_modules/parcel/lib/cli.js:295:21) {
  errno: -5,
  code: 'EIO',
  syscall: 'setRawMode'
}
^C

user.com on  master is 📦 v5.11.0 via  v18.8.0
❯ ls
```

If we look closely at the second prompt. After first press of `Ctrl-c`. The background process of Parcel printing `Parcel is shutting down...` over and over again until I press `Ctrl-c` again. Then I can use `ls` as normal.

In nushell, the first `Ctrl-c` doesn't produce the similar behavior. No `Parcel is shutting down...`. and the terminal just freeze.

*Hopefully, It add the insight to the maintainer*.

**Comment by evelant** [comment]
Is there any more information I could provide to help triage this? I can't really use nushell as my daily driver because of this bug. It hangs every time I need to stop and restart my dev tools.

**Comment by fdncred** [comment]
@evelant have you tried the latest release 0.70?

**Comment by evelant** [comment]
@fdncred I just updated to latest and it is still a problem

**Comment by fdncred** [comment]
I'm not sure what is causing the problem nor how to fix it.

**Comment by cacilhas** [comment]
I got a very similar issue, but Nushell seems to freeze randomly. Sometimes it freezes just starting!! Too many times indeed…

I have Nushell 0.76.0 install in ArchLinux, and in macOS. This issue doesn’t happen on macOS, only on Linux.

**Comment by sholderbach** [comment]
Is this still an issue after the signal and process group fixes by @IanManske ?

**Comment by 0x4d6165** [comment]
Still getting this issue in 0.97.1 on arch

**Comment by cptpiepmatz** [comment]
Closing this as we don't have any way to solve this without a reproducible setup.

**Comment by margaretjoanmiller** [comment]
@cptpiepmatz what would be needed to reproduce it? just a config? I can provide mine if that would be helpful

**Comment by cptpiepmatz** [comment]
Do you still have this issue with the latest version of nu?

**Comment by margaretjoanmiller** [comment]
Last I checked which was a few weeks ago but I can try on the latest in case I missed a version

EDIT: Actually, I wonder if it's related to any of the three main things I add to my shell configs: zoxide, starship prompt, and atuin. I wonder if any of them are causing the issue at least for myself

**Comment by cptpiepmatz** [comment]
Could you try to see if this still happens when you use `nu --no-config-file`, this could help us narrow it down.

**Comment by margaretjoanmiller** [comment]
I would except I had to move to windows for a class (ugh) and I've found nu inserts a bunch of new lines on every key press on WSL? will investigate more

**Comment by cptpiepmatz** [comment]
That sounds like your terminal emulator doesn't support one of the osc codes, wezterm for example has trouble with it.

**Comment by margaretjoanmiller** [comment]
Seems like it is indeed one of those three plugins, a clean config does not have this issue.

**Comment by cptpiepmatz** [comment]
Try to reproduce this with a minimal setup and open a fresh issue

**Comment by margaretjoanmiller** [comment]
I actually can't reproduce it now so something must've changed! My bad!
